src/buffer/out/TextAttribute.hpp

```diff
@@ -163,12 +163,36 @@ class TextAttribute final
         return _foreground.IsRgb() || _background.IsRgb();
     }
 
+    // In certain scenarios, we don't care about specifically the foreground color.
+    constexpr bool EqualsExceptForeground(const TextAttribute& other, const bool inverted = false) const noexcept
+    {
+        // sneaky-sneaky: I'm using xor here
+        // inverted is whether there's a global invert; Reverse is a local one.
+        // global ^ local == true : the background attribute is actually the visible foreground, so we care about the foregrounds being identical
+        // global ^ local == false: the foreground attribute is the visible foreground, so we care about the backgrounds being identical
+        const auto checkForeground = (inverted != _IsReverseVideo());
+        return (_wAttrLegacy & META_ATTRS) == (other._wAttrLegacy & META_ATTRS) &&
+               ((checkForeground && _foreground == other._foreground) ||
+                (!checkForeground && _background == other._background)) &&
+               _extendedAttrs == other._extendedAttrs;
+    }
+
+    constexpr bool IsAnyGridLineEnabled() const noexcept
+    {
+        return WI_IsAnyFlagSet(_wAttrLegacy, COMMON_LVB_GRID_HORIZONTAL | COMMON_LVB_GRID_LVERTICAL | COMMON_LVB_GRID_RVERTICAL | COMMON_LVB_UNDERSCORE);

```

Comment:
Do we want to future proof this by checking the `ExtendedAttributes` too (CrossedOut, Underlined, etc)? It's a bit open ended, because there are all sorts of these attributes that we may want to implement one day, so I don't know. I just thought it worth mentioning.

Reply:
you're not wrong, I just keep acting like i've never written code before. :smile:

Reply:
I love this thread you guys